### PR TITLE
argocd 1.6.1 (new formula)

### PR DESCRIPTION
**Pull request by crunchtime-ali**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Here is the CD CLI matching the PR for Argo CLI itself https://github.com/Homebrew/homebrew-core/pull/58259

### CODE REVIEW

File: Formula/argocd.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      :revision => "159674ee844a378fb98fe297006bf7b83a6e32d2"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    inreplace "Makefile", "CGO_ENABLED=0", ""
+    system "make", "cli"
+    bin.install "dist/argocd"
+    prefix.install_metafiles
```
Comment (dawidd6) [inline Formula/argocd.rb:15]:
```suggestion
```

File: Formula/argocd.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://argoproj.io"
+  url "https://github.com/argoproj/argo-cd.git",
+      :tag      => "v1.6.1",
+      :revision => "159674ee844a378fb98fe297006bf7b83a6e32d2"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    inreplace "Makefile", "CGO_ENABLED=0", ""
```
Comment (dawidd6) [inline Formula/argocd.rb:12]: Why this change?

Comment (crunchtime-ali) [reply Formula/argocd.rb:12]: Same problem as with the `argo` formula. It fails with these errors:
```
go build os/user: copying /private/tmp/go-build946619233/b142/pkg.a: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/os/user.a: operation not permitted
go build net: copying /private/tmp/go-build946619233/b074/pkg.a: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/net.a: operation not permitted
```

**Comment by dawidd6** [comment]
Squash commits and it will be good to go I think.

**Review by dawidd6 — DISMISSED** [review]

**Comment by crunchtime-ali** [comment]
@dawidd6 Can you kick the CI server? CI runs for 10.14 and 10.14 seem to be queued forever.

**Comment by dtrodrigues** [comment]
CI is running again now for this.

**Comment by crunchtime-ali** [comment]
> CI is running again now for this.

Thanks it successfully finished now. Why isn't the merge-bot picking it up?

**Comment by SMillerDev** [comment]
Cause it's a new formula. Those need to be merged manually.

File: Formula/argocd.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/argocd.rb:12]: might be good to have this as comment for future reference.

**Review by chenrui333 — DISMISSED** [review]
lgtm

File: Formula/argocd.rb (hunk 2 above)
Comment (crunchtime-ali) [reply Formula/argocd.rb:13]: @chenrui333 I added a comment
